cms_coding_rules.md

```diff
@@ -0,0 +1,144 @@
+# CMS Naming, Coding, And Style Rules
+
+### By the CMS Ofﬂine Software Development Team
+
+## Abstract
+This document describes the design, naming, coding, and style rules and recommendations for CMS software written in C++, plus a few guidelines for Python configuration.
+
+
+## Outline
+#### [1 -- Introduction](#1----introduction-1)
+#### [2 -- Naming Rules](#2----naming-rules-1)
+#### [3 -- Style Rules](#3----style-rules-1)
+#### [4 -- Technical Coding Rules](#4----technical-coding-rules-1)
+#### [5 -- Documentation Rules](#5----documentation-rules-1)
+#### [6 -- Packaging Rules](#6----packaging-rules-1)
+#### [7 -- Design and Coding Guidelines](#7----design-and-coding-guidelines-1)
+
+## 1 -- Introduction
+This document describes the CMS C++ software naming, coding, style, and documentation rules and recommendations.
+All CMS C++ software is expected to comply with the rules. The asterisk (\*) after some rules indicates that there may be exceptional use cases where the rule may be violated with good justiﬁcation.
+Coding rules are meant to prevent serious problems in software function, performance, maintainability, usability, and portability. The Packaging Rules section also has some brief guidelines for Python configuration.
+
+## 2 -- Naming Rules
+1. C++ header ﬁles use the sufﬁx .h, e.g. `CaloCluster.h`. (\*)
+2. For C++ source ﬁles, the preferred sufﬁx  is .cc, e.g. `CaloCluster.cc`, though there are existing files using .cpp and .cxx suffixes. (\*)
```

Comment:
I would suggest to rename all `.cxx` files to `.cc` files and drop their mention here.

Reply:
To simplify the rule, I removed mention of cpp and cxx and left the exception indicator. We can hope those suffixes get changed to .cc soon.